pkg/endpoint/policy.go

```diff
@@ -311,41 +309,61 @@ func (e *Endpoint) regenerate(owner Owner) error {
 	return nil
 }
 
-// Force regeneration of endpoint programs & policy
-func (e *Endpoint) regenerateLocked(owner Owner) error {
-	err := e.regenerate(owner)
-	if err != nil {
-		e.LogStatus(BPF, Failure, err.Error())
-	} else {
-		e.LogStatusOK(BPF, "Successfully regenerated endpoint program")
-	}
-
-	return err
-}
-
-// Regenerate forces the regeneration of endpoint programs & policy.
-func (e *Endpoint) Regenerate(owner Owner) error {
-	return e.regenerateLocked(owner)
+// Regenerate forces the regeneration of endpoint programs & policy
+func (e *Endpoint) Regenerate(owner Owner) <-chan bool {
+	newReq := &Request{
+		ID:           e.ID,
+		MyTurn:       make(chan bool),
+		Done:         make(chan bool),
+		ExternalDone: make(chan bool),
+	}
+	owner.QueueEndpoint(newReq)
+	go func(e *Endpoint, myTurn <-chan bool, finish chan<- bool, externalFinish chan<- bool) {
```

Comment:
Should this goroutine be cancelled if the endpoint is deleted before the build happens?